go-controller/pkg/ovn/controller/apbroute/master_controller.go

```diff
@@ -508,40 +520,33 @@ func (c *ExternalGatewayMasterController) processNextPodWorkItem(wg *sync.WaitGr
 }
 
 // updateStatusAPBExternalRoute updates the CR with the current status of the CR instance, including errors captured while processing the CR during its lifetime
-func (c *ExternalGatewayMasterController) updateStatusAPBExternalRoute(externalRoutePolicy *adminpolicybasedrouteapi.AdminPolicyBasedExternalRoute, processedError error) error {
-	if externalRoutePolicy == nil {
-		// policy doesnt exist anymore, nothing to do
+func (c *ExternalGatewayMasterController) updateStatusAPBExternalRoute(policyName string, gwIPs sets.Set[string],
+	processedError error) error {
+	if gwIPs == nil {
+		// policy doesn't exist anymore, nothing to do
 		return nil
```

Comment:
I changed it to return its own error on purpose, so that we know the result of the status update itself, and the caller  already knows the processedError, I store these in separate variables to avoid confusion.
e.g in the existing code before this change, failed delete will not be retried, because `updateStatusAPBExternalRoute` never returns error for deleted policy https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/controller/apbroute/master_controller.go#L512C12-L514
also status update failure should probably just be logged instead of causing the whole policy resync